crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -190,6 +190,28 @@ impl Order {
                 .unwrap_or_default()
                 .into();
 
+        // adjust ammounts if order has volume based protocol fee
```

Comment:
Can you reference the issue here that will allow solvers to skip all driver fee adjustments (so we can more easily find the places that need adjustments). 

Also are you sure this is the right place to add this logic? Won't available balance smaller than the full balance cause issues for fill or kill orders (an e2e test should tell us)?

Reply:
> Won't available balance smaller than the full balance cause issues for fill or kill orders (an e2e test should tell us)

If you refer to function `filter_orders` where we filter orders with insufficient balance, then AFAIS this change affects only the BUY orders and it lowers the required sell amount which means filtering is less strict than it was before. Although, truly introduces imprecision so I will move it out. 

If you refer to fulfillments requirement where we require solvers to return executed amount and fees which are in total equal to sell amount for sell orders, I don't think this affects this logic since it works with surplus token and not with target token. 

Will add tests to confirm this.

Reply:
Moved calculation out. Added tests.